pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -946,6 +1027,11 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 
 	if keepCount == 0 && len(changes.ContainersToStart) == 0 {
 		changes.KillPod = true
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			if len(changes.ContainersToRestart) != 0 {
```

Comment:
move this condition up one level

Reply:
you mean move it into the same line as featuregate check? if yes, this is by intention- typically when we get rid of featuregates, it is easy to just remove the featuregate line and not have to rewrite the code. but i can move it up if you feel strongly about it.

Reply:
keeping this so that feature-gate flag removal is simpler.